tfjs-backend-webgpu/src/webgpu_util.ts

```diff
@@ -56,6 +58,24 @@ export function computeDispatch(
                    (workGroupSize[2] * elementsPerThread[2])) :
                1
   ];
+
+  if (dispatchX <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchY <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE &&
+      dispatchZ <= MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE) {
+    return [dispatchX, dispatchY, dispatchZ];
+  }
+
+  util.assert(dispatchX > MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE && !layout.y
+      && !layout.z, () => `Only support flatted layout to reshape dispatch ` +
+      `size when the dispatch size exceeds the limit size on x dimension.`);
```

Comment:
Seems like "flattening" should always be in the X direction. If the dispatchY or dispatchZ is too large, I'd guess that might not be a flattened tensor but too large for some other reason? So how about making this message more generic:

```suggestion
  util.assert(dispatchX > MAX_COMPUTE_PER_DIMENSION_DISPATCH_SIZE && !layout.y
      && !layout.z, () => 'Dispatch size exceeds WebGPU limits in Y or Z dimension.');
```